### PR TITLE
Initial solution to assignment 6

### PR DESCRIPTION
**Pull request by comRamona**
During our initial meeting, we agreed on a couple of points. I have tried to address most of them in the initial solution.

- idea: in the current free space, move there the car that should be in that space in the final state, until the current free space is equal to the final free space.
- initial check if initialFreeSpace == finalFreeSpace (so final state is a permutation of initial state) then firstly do a dummy move of a car that isn't in its final space to initialFreeSpace. -- ->DID NOT DO THIS
- for input, we decided to go with 2 maps of N capacity for initial and final state, mapping each space to its car.
- key (space) and value (carId) of maps can be Integers.
- we will keep a currentFreeSpace variable to keep track of the current not occupied space.
- we will create a Move object with from, to and probably car/carId fields and boolean isValid() method to represent a move. ----> DID NOT DO THIS
- return a list of Moves. 
- we can also create the two equivalents maps of car to its current space, we will experiment/discuss this approach in detail during implementation.
- tests will check if moves are valid and lead to the desired final state.  DID NOT CHECK for valid moves, just for final state, I think there is no need for the validity check, given that I validate the input and end state and given that the swapping is pretty straightforward

### CODE REVIEW

File: group/assignment6/ParkingLot.java [hunk 1]
```diff
@@ -26,6 +25,29 @@ public ParkingLot(int[] parkingLotCurrent, int[] parkingLotEnd) {
...
         moves = new ArrayList<>();
+        if (initialEqualsFinalFreeSpace(parkingLotCurrent, parkingLotEnd)) {
+        	for (int i = 0; i < N-1; i++) {
+                if (parkingLotCurrent[i] == -1) {
+                    if (i == 0) {
+                    	int counter = i+1;
+                    	while (counter<N-1 && parkingLotCurrent[counter] == parkingLotEnd[counter])
+                    	{counter++;}
+                    	if (counter!=N-1) {
+                    		int temp = parkingLotCurrent[i];
```
Comment (comRamona) [inline group/assignment6/ParkingLot.java:36]: If you are swapping cars, shouldn't these be added to the moves array?

File: group/assignment6/ParkingLot.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+            }
+        }
+        for (String move : moves) {
+            System.out.println(move);
+        }
+    }
+
+    public static void checkValidInput(int[] start, int[] end) {
+        assertEquals(start.length, end.length);
```
Comment (dianatanase) [inline group/assignment6/ParkingLot.java:91]: Should we add errors rather than assertions?

File: group/assignment6/ParkingLot.java [hunk 3]
```diff
@@ -82,6 +104,25 @@ public static void checkValidInput(int[] start, int[] end) {
...
+        int count = 0;
+        for (int i = 0; i < end.length; i++) {
+            if (start[i] == end[i]) {
+            	count++;
+            }
+        }
+        if (count==end.length-1) {
+        	assertFalse("No moves were needed.",found[count + 1]);
+        }
+    }
```
Comment (dianatanase) [inline group/assignment6/ParkingLot.java:116]: I wouldn't say that if the initial and final states are equal then the input is not valid, better return an empty list as no moves are needed.

File: group/assignment6/ParkingLot.java (hunk 1 above)
Comment (dianatanase) [reply group/assignment6/ParkingLot.java:36]: Why swap cars, which involves 2 moves, instead of moves only one car to the free space?

**Pull request closed without merging**